### ISSUE TITLE
App Session events don't seem to be logged after v1.2.4

### ISSUE THREAD

**Issue by mark-carrington** [mixpanel/mixpanel-react-native#91]
Since upgrading mixpanel-react-native from v1.1.1 to v1.3.1 we've noticed that 'App Session' events seem to no longer be logged in MixPanel. I tried downgrading through some of the recent versions, and noticed that on v1.2.4 the 'App Session' events still seem to be logged. But on v1.3.0 and upwards they no longer seem to show up.

As part of the upgrade we changed how we initialise MixPanel, so we now use the instance method. All other events seem to be coming through to the dashboard just fine.

Any ideas what might be causing this?

**Comment by zihejia** [comment]
hi @mark-carrington , looks like there are some problems for React Native code in sync with our embedded native iOS SDK, we will address this issue for the coming release.

**Comment by mark-carrington** [comment]
> hi @mark-carrington , looks like there are some problems for React Native code in sync with our embedded native iOS SDK, we will address this issue for the coming release.

Thanks for looking into this @zihejia! Is there an ETA for the fix?

**Comment by andyPandy1337** [comment]
I've also encountered this problem. Not sure, but it seems since [this commit](https://github.com/mixpanel/mixpanel-react-native/commit/6c03deb3aefd68e540ddb70c8b54e5d2b5929cae) ensure automatic tracking is always disabled for iOS?

Based on limited testing, it seems changing `instance?.trackAutomaticEventsEnabled = true`  makes the App Session events work again. But unclear if there are any downstream consequences of doing so. Hoping for a proper fix soon.

```swift
func initialize(_ token: String,
                    optOutTrackingByDefault: Bool = false,
                    properties: [String: Any],
                    resolver resolve: RCTPromiseResolveBlock,
                    rejecter reject: RCTPromiseRejectBlock) -> Void {
        AutomaticProperties.setAutomaticProperties(properties)
        Mixpanel.initialize(token: token, launchOptions: nil, flushInterval: Constants.DEFAULT_FLUSH_INTERVAL, instanceName: token, automaticPushTracking: Constants.AUTOMATIC_PUSH_TRACKING, optOutTrackingByDefault: optOutTrackingByDefault)
        let instance = MixpanelReactNative.getMixpanelInstance(token)
        // Why always set to false here?
        instance?.trackAutomaticEventsEnabled = false
        resolve(true)
    }
```

**Comment by Haarmindeer** [comment]
I am also facing the same issue. Tracking for IOS is not working at my side, but in case of android it works perfectly fine. I tried to log for iOS its just give me  the following log

[Mixpanel - MixpanelInstance.swift - func loggingEnabled] (info) - Logging Enabled

**Comment by Haarmindeer** [comment]
I am also facing the same issue. Tracking for IOS is not working on my side, but it works perfectly fine in the case of Android. I tried to log for iOS it's just giving me the following logs 

[Mixpanel - MixpanelInstance.swift - func loggingEnabled] (info) - Logging Enabled

**Comment by zihejia** [comment]
it's been fixed in [1.3.3](https://github.com/mixpanel/mixpanel-react-native/releases/tag/v1.3.3), closing this one now.
